diff --git a/gcc/gcc/c-family/c-lex.c b/gcc/gcc/c-family/c-lex.c
--- a/gcc/gcc/c-family/c-lex.c
+++ b/gcc/gcc/c-family/c-lex.c
@@ -1353,10 +1353,15 @@ lex_charconst (const cpp_token *token)
     }
   /* In C, a character constant has type 'int'.
      In C++ 'char', but multi-char charconsts have type 'int'.  */
-  else if (!c_dialect_cxx () || chars_seen > 1)
-    type = integer_type_node;
-  else
+  else if (c_dialect_cxx ())
     type = char_type_node;
+  else if (chars_seen > 2)
+    type = long_unsigned_type_node;
+  else if (chars_seen > 1)
+    type = unsigned_type_node;
+  else
+    type = integer_type_node;
+
 
   /* Cast to cppchar_signed_t to get correct sign-extension of RESULT
      before possibly widening to HOST_WIDE_INT for build_int_cst.  */
diff --git a/gcc/libcpp/charset.c b/gcc/libcpp/charset.c
--- a/gcc/libcpp/charset.c
+++ b/gcc/libcpp/charset.c
@@ -1797,6 +1797,6 @@ narrow_str_to_charconst (cpp_reader *pfile, cpp_string str,
 {
   size_t width = CPP_OPTION (pfile, char_precision);
-  size_t max_chars = CPP_OPTION (pfile, int_precision) / width;
+  size_t max_chars = 4;
   size_t mask = width_to_mask (width);
   size_t i;
   cppchar_t result, c;
@@ -1834,5 +1834,5 @@ narrow_str_to_charconst (cpp_reader *pfile, cpp_string str,
-  /* Multichar constants are of type int and therefore signed.  */
+  /* Palm FourCC constants are unsigned.  */
   if (i > 1)
-    unsigned_p = 0;
+    unsigned_p = 1;
   else
     unsigned_p = CPP_OPTION (pfile, unsigned_char);
@@ -1841,8 +1841,10 @@ narrow_str_to_charconst (cpp_reader *pfile, cpp_string str,
      sign- or zero-extend to the full width of cppchar_t.
      For single-character constants, the value is WIDTH bits wide.
      For multi-character constants, the value is INT_PRECISION bits wide.  */
-  if (i > 1)
-    width = CPP_OPTION (pfile, int_precision);
+  if (i > 2)
+    width = 32;
+  else if (i > 1)
+    width = 16;
   if (width < BITS_PER_CPPCHAR_T)
     {
       mask = ((cppchar_t) 1 << width) - 1;
