diff --git a/gcc/zlib/zutil.h b/gcc/zlib/zutil.h
--- a/gcc/zlib/zutil.h
+++ b/gcc/zlib/zutil.h
@@ -132,15 +132,6 @@ extern z_const char * const z_errmsg[10]; /* indexed by 2-zlib_error */
 
 #if defined(MACOS) || defined(TARGET_OS_MAC)
 #  define OS_CODE  7
-#  ifndef Z_SOLO
-#    if defined(__MWERKS__) && __dest_os != __be_os && __dest_os != __win32_os
-#      include <unix.h> /* for fdopen */
-#    else
-#      ifndef fdopen
-#        define fdopen(fd,mode) NULL /* No fdopen() */
-#      endif
-#    endif
-#  endif
 #endif
 
 #ifdef __acorn
diff --git a/gcc/gcc/config/host-darwin.c b/gcc/gcc/config/host-darwin.c
--- a/gcc/gcc/config/host-darwin.c
+++ b/gcc/gcc/config/host-darwin.c
@@ -20,9 +20,13 @@
 #include "config.h"
 #include "system.h"
 #include "coretypes.h"
+#include "hosthooks.h"
+#include "hosthooks-def.h"
 #include "diagnostic-core.h"
 #include "config/host-darwin.h"
 
+const struct host_hooks host_hooks = HOST_HOOKS_INITIALIZER;
+
 /* Yes, this is really supposed to work.  */
 static char pch_address_space[1024*1024*1024] __attribute__((aligned (4096)));
 
diff --git a/gcc/gcc/system.h b/gcc/gcc/system.h
--- a/gcc/gcc/system.h
+++ b/gcc/gcc/system.h
@@ -201,6 +201,24 @@ extern int fprintf_unlocked (FILE *, const char *, ...);
 #ifdef INCLUDE_STRING
 # include <string>
 #endif
+#if defined (INCLUDE_ALGORITHM) || !defined (HAVE_SWAP_IN_UTILITY)
+# include <algorithm>
+#endif
+#ifdef INCLUDE_LIST
+# include <list>
+#endif
+#ifdef INCLUDE_MAP
+# include <map>
+#endif
+#ifdef INCLUDE_SET
+# include <set>
+#endif
+#ifdef INCLUDE_VECTOR
+# include <vector>
+#endif
+# include <cstring>
+# include <new>
+# include <utility>
 #endif
 
 /* There are an extraordinary number of issues with <ctype.h>.
