diff --git a/gcc/libgcc/config.host b/gcc/libgcc/config.host
--- a/gcc/libgcc/config.host
+++ b/gcc/libgcc/config.host
@@ -1019,7 +1019,7 @@ m32rle-*-elf*)
 	tmake_file=t-fdpbit
 	;;
 m68k-*-elf* | fido-*-elf)
-	tmake_file="$tmake_file m68k/t-floatlib"
+	tmake_file="$tmake_file m68k/t-floatlib m68k/t-palmos-pic"
 	;;
 m68k-*-macos)
 	extra_parts="crtbegin.o crtend.o"
diff --git a/gcc/libgcc/config/m68k/t-palmos-pic b/gcc/libgcc/config/m68k/t-palmos-pic
new file mode 100644
--- /dev/null
+++ b/gcc/libgcc/config/m68k/t-palmos-pic
@@ -0,0 +1 @@
+LIBGCC2_CFLAGS += -mpcrel -fpic -O1
diff --git a/gcc/libgcc/config/m68k/lb1sf68.S b/gcc/libgcc/config/m68k/lb1sf68.S
--- a/gcc/libgcc/config/m68k/lb1sf68.S
+++ b/gcc/libgcc/config/m68k/lb1sf68.S
@@ -104,11 +104,11 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 	/* Non PIC (absolute/relocatable) versions */
 
 	.macro PICCALL addr
-	jbsr	\addr
+	bsr	\addr
 	.endm
 
 	.macro PICJUMP addr
-	jmp	\addr
+	bra	\addr
 	.endm
 
 	.macro PICLEA sym, reg
