diff --git a/gcc/libgcc/Makefile.in b/gcc/libgcc/Makefile.in
--- a/gcc/libgcc/Makefile.in
+++ b/gcc/libgcc/Makefile.in
@@ -247,7 +247,7 @@ LIBGCC2_CFLAGS = -O2 $(LIBGCC2_INCLUDES) $(GCC_CFLAGS) $(HOST_LIBGCC2_CFLAGS) \
 		 $(LIBGCC2_DEBUG_CFLAGS) -DIN_LIBGCC2 \
 		 -fbuilding-libgcc -fno-stack-protector \
 		 -ffunction-sections \
-		 $(INHIBIT_LIBC_CFLAGS)
+		 $(INHIBIT_LIBC_CFLAGS) -mpcrel -fpic -O1
 
 # Additional options to use when compiling libgcc2.a.
 # Some targets override this to -isystem include
diff --git a/gcc/libgcc/config/m68k/lb1sf68.S b/gcc/libgcc/config/m68k/lb1sf68.S
--- a/gcc/libgcc/config/m68k/lb1sf68.S
+++ b/gcc/libgcc/config/m68k/lb1sf68.S
@@ -104,11 +104,11 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 	/* Non PIC (absolute/relocatable) versions */
 
 	.macro PICCALL addr
-	jbsr	\addr
+	bsr	\addr
 	.endm
 
 	.macro PICJUMP addr
-	jmp	\addr
+	bra	\addr
 	.endm
 
 	.macro PICLEA sym, reg
